### PR TITLE
Fixed Undead units bleeding

### PR DESCRIPTION
**Pull request by tristanka**
Patched the agent particle logic in Mission to prevent Undead units from
spraying blood on hit.

### CODE REVIEW

File: Abilities/AbilityComponent.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public AbilityComponent(Agent agent) : base(agent)
 
         public void SelectAbility(int index)
         {
-            if (_knownAbilities.Count > 0)
+            if (_knownAbilities.Count > 0 && index >= 0)
```
Comment (tristanka) [inline Abilities/AbilityComponent.cs:51]: If index was somehow less than 0, the index could end up negative as well, so this ensures we avoid that.

File: Abilities/FireBallAbilityScript.cs [hunk 2]
```diff
@@ -96,15 +96,15 @@ protected override void OnPhysicsCollision(ref PhysicsContact contact)
...
+                var collisionNormal = contact.ContactPair0.Contact0.Normal;
                 //apply AOE damage
-                TOWBattleUtilities.DamageAgentsInArea(collisionpoint.AsVec2, _radius, _minDamage, _maxDamage, _casterAgent, true);
+                TOWBattleUtilities.DamageAgentsInArea(collisionPoint.AsVec2, _radius, _minDamage, _maxDamage, _casterAgent, true);
                 //create visual explosion
                 var explosion = GameEntity.CreateEmpty(Scene);
                 MatrixFrame frame = MatrixFrame.Identity;
                 var psys = ParticleSystem.CreateParticleSystemAttachedToEntity("psys_burning_projectile_default_coll", explosion, ref frame);
-                var globalFrame = new MatrixFrame(Mat3.CreateMat3WithForward(in collisionnormal), collisionpoint);
+                var globalFrame = new MatrixFrame(Mat3.CreateMat3WithForward(in collisionNormal), collisionPoint);
```
Comment (tristanka) [inline Abilities/FireBallAbilityScript.cs:107]: In this file, I just camel cased some variables.

**Review by Z3rca — APPROVED** [review]
fairly understandable also for me :)

**Review by hunharibo — APPROVED** [review]
Nice work, approved!

**Pull request merged**